gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ManagedController.java

```diff
@@ -128,6 +133,29 @@ DeliveryConfig deleteManifest(@PathVariable("name") String name) {
     return keelService.deleteManifest(name);
   }
 
+  @ApiOperation(value = "Validate a delivery config manifest", response = Map.class)
+  @PostMapping(
+      path = "/delivery-configs/validate",
+      consumes = {APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE},
+      produces = {APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE})
+  ResponseEntity<Map> validateManifest(@RequestBody DeliveryConfig manifest) {
+    try {
+      return ResponseEntity.ok(keelService.validateManifest(manifest));
+    } catch (RetrofitError e) {
+      if (e.getResponse().getStatus() == 400) {
+        try {
+          return ResponseEntity.badRequest()
+              .body(objectMapper.readValue(e.getResponse().getBody().in(), Map.class));
```

Comment:
This ensures that the detailed error information returned by keel is included in the response from gate.